src/pages/RepoPage/FailedTestsTab/hooks/useInfiniteTestResults.tsx

```diff
@@ -0,0 +1,208 @@
+import {
+  useInfiniteQuery,
+  type UseInfiniteQueryOptions,
+} from '@tanstack/react-query'
+import { z } from 'zod'
+
+import {
+  RepoNotFoundErrorSchema,
+  RepoOwnerNotActivatedErrorSchema,
+} from 'services/repo'
+import Api from 'shared/api'
+import { type NetworkErrorObject } from 'shared/api/helpers'
+import A from 'ui/A'
+
+const TestResultSchema = z.object({
+  updatedAt: z.string(),
+  name: z.string(),
+  commitsFailed: z.number().nullable(),
+  failureRate: z.number().nullable(),
+  avgDuration: z.number().nullable(),
+})
+
+type TestResult = z.infer<typeof TestResultSchema>
+
+const GetTestResultsSchema = z.object({
+  owner: z
+    .object({
+      repository: z.discriminatedUnion('__typename', [
+        z.object({
+          __typename: z.literal('Repository'),
+          testResults: z.object({
+            edges: z.array(
+              z.object({
+                node: TestResultSchema,
+              })
+            ),
+            pageInfo: z.object({
+              endCursor: z.string().nullable(),
+              hasNextPage: z.boolean(),
+            }),
+          }),
+        }),
+        RepoNotFoundErrorSchema,
+        RepoOwnerNotActivatedErrorSchema,
+      ]),
+    })
+    .nullable(),
+})
+
+const query = `
+query GetTestResults(
+  $owner: String!
+  $repo: String!
+  $filters: TestResultsFilters
+  $first: Int
+  $after: String
+  $last: Int
+  $before: String
+) {
+  owner(username: $owner) {
+    repository: repository(name: $repo) {
+      __typename
+      ... on Repository {
+        testResults(filters: $filters, first: $first, after: $after, last: $last, before: $before) {
+          edges {
+            node {
+              updatedAt
+              avgDuration
+              name
+              failureRate
+              commitsFailed
+            }
+          }
+          pageInfo {
+            endCursor
+            hasNextPage
+          }
+        }
+      }
+      ... on NotFoundError {
+        message
+      }
+      ... on OwnerNotActivatedError {
+        message
+      }
+    }
+  }
+}`
```

Comment:
`l`: Just tidying up the testResults args formatting

```suggestion
const query = `
query GetTestResults(
  $owner: String!
  $repo: String!
  $filters: TestResultsFilters
  $first: Int
  $after: String
  $last: Int
  $before: String
) {
  owner(username: $owner) {
    repository: repository(name: $repo) {
      __typename
      ... on Repository {
        testResults(
          filters: $filters
          first: $first
          after: $after
          last: $last
          before: $before
        ) {
          edges {
            node {
              updatedAt
              avgDuration
              name
              failureRate
              commitsFailed
            }
          }
          pageInfo {
            endCursor
            hasNextPage
          }
        }
      }
      ... on NotFoundError {
        message
      }
      ... on OwnerNotActivatedError {
        message
      }
    }
  }
}`
```